CMakePresets.json

```diff
@@ -0,0 +1,16 @@
+{
+  "version": 3,
+  "configurePresets":[
+    {
+      "name": "dev",
+      "binaryDir": "build",
+      "installDir": "${sourceDir}/build/install",
```

Comment:
Isn't installing inside the build directory rather bad practice?

Reply:
I've never heard anyone discourage it or argue against doing so for developer builds.

It's great for development because it means we can install SFML without needing root access and you don't run the risk of polluting your machine with potentially bugged installs of SFML. Plus it means you don't need to add yet another directory to your .gitignore. All build products remain in the binary directory which is already ignored. When you want to delete your build dir you also delete this install dir so it won't stick around polluting anything.

Reply:
Feels a bit weird, but I guess we can keep it for now